logstash-core-event/lib/logstash/event.rb

```diff
@@ -98,7 +98,13 @@ def clone
       copy[k] = begin v.clone rescue v end
     end
 
-    self.class.new(copy)
+    cloned = self.class.new(copy)
+    
+    # Clone metadata too
+    @metadata.each do |k,v|
+      cloned["@metadata"][k] = begin v.clone rescue v end
+    end
+    cloned
```

Comment:
I took a bit of time to go through the code I don't think we need the loop in that case.
I think this code should be enough in most case.

```
self.class.new(copy.merge({ "@metadata" => LogStash::Util.deep_clone(@metadata.clone) }))
```

It really handle recursive cloning of the values, we might want to use the same behavior for the `@data`